presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1679,6 +1680,42 @@ protected Scope visitExcept(Except node, Optional<Scope> scope)
             return visitSetOperation(node, scope);
         }
 
+        private boolean checkIfRelationUsedInExpressionJointlyWithOtherRelation(Expression expression, Relation relation)
+        {
+            if (expression instanceof LogicalBinaryExpression) {
+                LogicalBinaryExpression logicalBinaryExpression = (LogicalBinaryExpression) expression;
+                switch (logicalBinaryExpression.getOperator()){
```

Comment:
switch looks redundant as you do the same thing in both cases.

Reply:
Typically, switch is preferred when working with enumerations as compiler can suggest when you forger to handle some of the enums. Here, the things are almost similar except the reduction operation that need to be handled differently:
1) || for AND 
2) && for OR

Reply:
You should not need different logic for AND/OR. Just not referencing in either is bad enough

Reply:
In fact, there is a difference. 
For query below, there should not be any warning 
`select * from t1 join t2 on t1.a = t2.a AND t2.b = 3`
However, for this query `select * from t1 join t2 on t1.a = t2.a AND t2.b = 3` it should be a performance warning.